daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -148,22 +168,26 @@ class VariableInstance private (
     this._value = Some(v)
   }
 
-  def setDefaultValue(v: DataValuePrimitiveNullable) = this._value = Some(v)
+  /* This is used to set a default value without assigning a priorValue */
+  def setDefaultValue(v: DataValuePrimitiveNullable) = {
+    Assert.invariant(this.state == VariableUndefined || this.state == VariableInProcess)
```

Comment:
Should we assert the ``v != NoValue`` too?

Reply:
Yes, done.